internal/handlers/deployments.go

```diff
@@ -163,6 +163,9 @@ func validateDeploymentRequest(
 		} else if *request.ApplicationLicenseID != *target.Deployment.ApplicationLicenseID {
 			return badRequestError(w, "can not update license")
 		}
+		if target.Deployment.ApplicationID != (*app).ID {
```

Comment:
I think this deref is not necessary
```suggestion
		if target.Deployment.ApplicationID != app.ID {
```